data-table/table.js

```diff
@@ -687,33 +686,100 @@ function createRowDeleteBtn(config) {
-        Reflect.deleteProperty(configDict, clientConfig.wrapperDivId);
-        dtCreateDataTable(clientConfig);
+    JSONBtn.onclick = function () {
+        // eslint-disable-next-line no-console
+        console.log(dtToJSON(clientConfig));
     }
-    wrapperDiv.appendChild(resetBtn);
+    wrapperDiv.appendChild(JSONBtn);
 }
 
-    /**
-// eslint-disable-next-line no-unused-vars
-/** function toJSON() {
-     * Parses data held in HTML to JSON and sends it to client
-     * TODO: Fill this out
-     * TODO: Implement Serialization
-} */
+/**
+ * Returns the HTML element corresponding to a cell at a specific index
+ * @param {object} config           - Table configuration object
+ * @param {Number} row              - Row on which the cell is located
+ * @param {Number} column           - Column in which the cell is located
+ * @returns {HTMLTableCellElement}  - HTML element of a specific cell
+ */
+function getCellElement(config, row, column) {
+    return document.getElementById(cellIndexToElementId(config.wrapperDivId, row, column))
+}
+
+/**
+ * Gets a 2D array containing all data from all fields of all cells in the table
+ * @param {object} config   - Table configuration object
+ * @returns {*[]}           - 2D array of cell objects
+ */
+function getTableData(config) {
+    let tableData = [];
+    for (let row = 0; row < config.numRows; row++) {
+        tableData.push(getRowData(config, row));
+    }
+    return tableData;
+}
+
+/**
+ * Gets a 1D array containing all data from all fields of all cells in a specific row of the table
+ * @param {object} config   - Table configuration object
+ * @param {Number} row      - Specific row to pull data from
+ * @returns {*[]}           - 1D array of cell objects
+ */
+function getRowData(config, row) {
+    let rowData = [];
+    for (let col = 1; col < config.numColumns; col++) {
+        rowData.push(getCellData(config, row, col));
+    }
+    return rowData
+}
+
+/**
+ * Gets an object containing all data from all fields of a specific cell
+ * @param {object} config   - Table configuration object
+ * @param {Number} row      - Row index for the cell
+ * @param {Number} col      - Column index for the cell
+ * @returns {object}        - Object containing cell data in the following format: {[fieldName]:[fieldValue],...}
+ */
+function getCellData(config, row, col) {
+    let cellData = {};
+    let cell = getCellElement(config, row, col);
+    let index = 0;
+
+    /*
+    Assumes that each cell has only field labels as children
+     */
+    cell.childNodes.forEach(function (label) {
+        /*
+        Assumes that each label has two children. The first is
+        text containing the label's name and the second is the
+        space the user can interact with.
+         */
+        let value = "";
+        switch (config.datumConfig.types[index]) {
+            case Number:
+                value = parseInt(label.childNodes[1].value, 10) || config.datumConfig.values[index];
```

Comment:
Perhaps we should force this to be filled out rather than requiring a default? It'll take some more thinking here...should all fields be mandatory? Or perhaps if it's empty, we send `null` and we let whoever is serializing it let us know of an error?

Doesn't need to be a part of this PR, but I am afraid of silently inserting default values. The UI should reflect the data that will be sent. Perhaps create a JIRA ticket for it + a TODO here?

Reply:
Okay, so I removed the `||` and the silent insertion of a default value. `value` is now set to `null` when `parseInt` encounters an empty string or for whatever reason can't parse the `String` into a `Number`.